### ISSUE TITLE
Missing styles with v3 and Safari 14

### ISSUE THREAD

**Issue by Artur-** [calebdwilliams/construct-style-sheets#84]
**Describe the bug**
The CSS rules in `adoptedStyleSheets` are supposed to end up inside `<style>` tags inside the shadow root. In Safari in some cases, the `<style>` tags are empty instead.

**To Reproduce**
https://start.vaadin.com/app/ is running the polyfill v2. If you open the URL in Safari, you can see a small globe icon next to "globe-solid" in the "Icon" select.

This comes from CSS which you can see by running
```
root = document.querySelector("view-item").shadowRoot;
styleSheets = root.adoptedStyleSheets;
cssRules = styleSheets.flatMap(sheet => Array.from(sheet.cssRules))
css = cssRules.map(cssRule => cssRule.cssText);
css.filter(text => text.includes(".la-globe"));
```

which shows that there is a `.la-globe` rule in the adopted style sheets array and correspondingly

```
Array.from(root.querySelectorAll("style")).filter(style => style.innerText.includes(".la-globe"))
```

shows there is a `.la-globe` rule in the style tags inside the shadow root.

Now I deployed a version of the app using v3 of the polyfill at https://start-dev.herokuapp.com/app/

There you cannot see a globe icon in the UI and running the same code reveals that `adoptedStylesheet` still contains the expected CSS but the style tags inside the shadow root does not contain the corresponding CSS rule. In fact most or all of the style tags are completely empty. 

This only reproduces in Safari.

In Firefox, the `<style>` tags actually are also empty but there the globe is shown in the UI, as expected. The inspector also shows that it finds the `.la-globe` CSS rule although it is unable to show where..

**Comment by Artur-** [comment]
I see, the styles are set using `style.sheet` in V3.

With V2, if you check the rule from the `<style>` tags using
```
Array.from(root.querySelectorAll("style")).flatMap(style => Array.from(style.sheet.cssRules)).filter(rule => rule.cssText.includes(".la-map::before")).map(rule => rule.cssText);
```
you get
```
[".la-map::before { content: ; }"]
```

With V3, the same query results in
```
[".la-map::before { }"]
```

So all is fine except that the `content: ;` part is completely missing

**Comment by Artur-** [comment]
The problem seems to be in this code in the polyfill
```
    function insertAllRules(from, to) {
        forEach.call(from.cssRules, function (rule, i) {
            to.insertRule(rule.cssText, i);
        });
    }
```

After running `to.insertRule(rule.cssText, i);` one would probably assume that
```
rule.cssText === to.rules[i].cssText
```

but for the content rules I see
```
> rule.cssText
< ".la-map::before { content: ; }"
> to.rules[i].cssText
< ".la-map::before { }"
```

**Comment by Lodin** [comment]
Confirmed. Looks like it is a Safari bug. I'll look for a workaround

**Comment by Lodin** [comment]
Found the reason. If you execute the following command:
```js
style.sheet.insertRule('.test { content: "foo" }');
```
And then try to get the value:
```js
console.log(style.sheet.cssRules[0].cssText);
```
You will get the following output:
```
.test { content: foo; } // no quotes
```
If you then try to add it to `insertRule`, nothing will be added:
```
.test { }
```
This happens only in Safari; other browsers (even IE11) handle this case correctly.

I guess it is connected with [this](https://bugs.webkit.org/show_bug.cgi?id=185953) bug. It also means that `cssText` in Safari is broken by design, and even if we get back to use `style.textContent` instead of CSS rules, we won't be able to support all the spec correctly.

The only choice I see here is to create patches for each broken rule like `content` and fix them manually, adding quotes or missing elements. However, that also means blowing the polyfill size up. @calebdwilliams, what do you think?

**Comment by calebdwilliams** [comment]
Well, I've gotten it working with a bit of hackery, I'm not sure if/when/how this would break though. The basic concept would be to manipulate the `rule.cssText` and look for `/content:(.*?);` and run add some the missing quotes like so:

```javascript
function insertAllRules(from, to) {
  forEach.call(from.cssRules, function(rule, i) {
    let cssText = rule.cssText;
    const isSafari = true;
    const content = /content:(.*?);/.exec(cssText);
    if (isSafari && content.length) {
      const value = cssText.replace(content[1], `"${content[1]}"`);
      cssText = value;
    }
    console.log(cssText, Date.now())
    to.insertRule(cssText, i);
  });
}
```

You can [see it working in this Pen](https://codepen.io/calebdwilliams/pen/mdwrzrj?editors=0010). This is the best solution I can think of for now, definitely open to other ideas. I don't really like the idea of re-implementing prototype rules (at all, but if we did, I would want that in a separate file from the core polyfill).

**Comment by Lodin** [comment]
Yes, I thought about a similar solution. However, there are pitfalls that I'm concerned about According to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/content), the `content` property can contain either enum values (`normal`, `none`, `open-quote`, etc.) and string values that can be anything. We cannot just wrap any value with quotes; we also have too little info about the value to perform the correct analysis. So I'm kinda confused.

Example of ambiguity:
```css
.test1 {
  content: normal;
}

.test2 {
  content: "normal";
}
```

**Comment by calebdwilliams** [comment]
Seems like the most effective solution then would be to run the above except for that set of conditions (which is fortunately small). I’d rather have a condition that developers can’t write `content: "normal"` than not use before or after. 

It’s either that or revert to the innerText rule that we used before.

**Comment by calebdwilliams** [comment]
@Artur- this should be fixed in the latest version, 3.0.3.
